src/bindings/python/src/pyopenvino/graph/passes/graph_rewrite.cpp

```diff
@@ -0,0 +1,75 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "pyopenvino/graph/passes/graph_rewrite.hpp"
+
+#include <pybind11/pybind11.h>
+
+#include <openvino/pass/graph_rewrite.hpp>
+#include <openvino/pass/pass.hpp>
+
+namespace py = pybind11;
+
+void regclass_passes_GraphRewrite(py::module m) {
+    py::class_<ov::pass::GraphRewrite, std::shared_ptr<ov::pass::GraphRewrite>, ov::pass::ModelPass, ov::pass::PassBase>
+        graph_rewrite(m, "GraphRewrite");
+    graph_rewrite.doc() =
+        "openvino.runtime.passes.GraphRewrite execute sequence of MatcherPass transformations in topological order";
+
+    graph_rewrite.def(py::init<>());
+    graph_rewrite.def(py::init([](const std::shared_ptr<ov::pass::MatcherPass>& pass) {
+                          return std::make_shared<ov::pass::GraphRewrite>(pass);
+                      }),
+                      py::arg("pass"),
+                      R"(
+                      Register single MatcherPass pass inside GraphRewrite.
+
+                      :param pass: MatcherPass instance
+                      :type pass: MatcherPass
+    // )");
+
+    graph_rewrite.def("add_matcher",
+                      static_cast<std::shared_ptr<ov::pass::MatcherPass> (ov::pass::GraphRewrite::*)(
+                          const std::shared_ptr<ov::pass::MatcherPass>&)>(&ov::pass::GraphRewrite::add_matcher),
+                      py::arg("pass"),
+                      R"(
+                      Register single MatcherPass pass inside GraphRewrite.
+
+                      :param pass: MatcherPass instance
+                      :type pass: MatcherPass
+    // )");
+
+    py::class_<ov::pass::BackwardGraphRewrite,
+               std::shared_ptr<ov::pass::BackwardGraphRewrite>,
+               ov::pass::GraphRewrite,
+               ov::pass::ModelPass,
+               ov::pass::PassBase>
+        back_graph_rewrite(m, "BackwardGraphRewrite");
+    back_graph_rewrite.doc() = "openvino.runtime.passes.BackwardGraphRewrite execute sequence of MatcherPass "
+                               "transformations in reversed topological order";
+
+    back_graph_rewrite.def(py::init<>());
+    back_graph_rewrite.def(py::init([](const std::shared_ptr<ov::pass::MatcherPass>& pass) {
+                               return std::make_shared<ov::pass::BackwardGraphRewrite>(pass);
+                           }),
+                           py::arg("pass"),
+                           R"(
+                           Register single MatcherPass pass inside BackwardGraphRewrite.
+
+                           :param pass: MatcherPass instance
+                           :type pass: MatcherPass
+    // )");
+
+    back_graph_rewrite.def(
+        "add_matcher",
+        static_cast<std::shared_ptr<ov::pass::MatcherPass> (ov::pass::BackwardGraphRewrite::*)(
+            const std::shared_ptr<ov::pass::MatcherPass>&)>(&ov::pass::BackwardGraphRewrite::add_matcher),
```

Comment:
```suggestion
        overload_cast_<const std::shared_ptr<ov::pass::MatcherPass>&>()&ov::pass::BackwardGraphRewrite::add_matcher),
```
Is it possible to implement in this way? 
For reference: https://pybind11.readthedocs.io/en/stable/classes.html#overloaded-methods

CC: @jiwaszki

Reply:
Overload cast works only with C++14, but we have only C++11.

Reply:
It is exactly an issue as Gleb stated. Generally function pointers are standard approach in pybind for C++11.

You might use an approach that is on the bottom of docs section. But AFAIK it wasn't working for us in all cases (_it might not be true after few releases of pybind and our code_), thus our code sticks to pointer casting.

Reply:
In short, I see no problem to keep the code as is.

Reply:
Tried to use `overload_cast_ ` here, but `overload_cast_impl` can't deduce return type for some reason.